### ISSUE TITLE
CheckChecking returns false-positives!

### ISSUE THREAD

**Issue by ArchieT** [ArchieT/3manchess#4]
```
 mf  ~  3manchess  game   checkmadesimple  go test -v
=== RUN   TestSimpleGenNoPanic
--- PASS: TestSimpleGenNoPanic (0.00s)
	move_test.go:8: first.After  Board:  [[WrN WnN WbN WqN WkN WbN WnN WrN GrN GnN GbN GqN GkN GbN GnN GrN BrN BnN BbN BqN BkN BbN BnN BrN] [#&# WpN WpN WpN WpN WpN WpN WpN GpN GpN GpN GpN GpN GpN GpN GpN BpN BpN BpN BpN BpN BpN BpN BpN] [#&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#] [WpN #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#] [#&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#] [#&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#]] [false false false] Gray [[true true] [true true] [true true]] [[127,127] [2,0]] 0 2 [true true true]
		
	move_test.go:10: Error first.After  We would be in check!
	move_test.go:14: second.After  <nil>
	move_test.go:16: Error second.After  That is not White's move, but Gray's
	move_test.go:20: third.After  <nil>
	move_test.go:22: Error third.After  How do you move that which does not exist?
	move_test.go:26: forth.After  Board:  [[WrN WnN WbN WqN WkN WbN WnN WrN GrN GnN GbN GqN GkN GbN GnN GrN BrN BnN BbN BqN BkN BbN BnN BrN] [#&# WpN WpN WpN WpN WpN WpN WpN #&# GpN GpN GpN GpN GpN GpN GpN BpN BpN BpN BpN BpN BpN BpN BpN] [#&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#] [WpN #&# #&# #&# #&# #&# #&# #&# GpN #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#] [#&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#] [#&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&# #&#]] [false false false] Black [[true true] [true true] [true true]] [[2,0] [2,8]] 0 3 [true true true]
		
	move_test.go:28: Error forth.After  We would be in check!
PASS
ok  	github.com/ArchieT/3manchess/game	0.003s
```

**Comment by ArchieT** [comment]
Fixed: dbf297a

**Comment by ArchieT** [comment]
Finally fixed in df157894272aa6108426b63685698e83d7c17ebb
